File: magic number.cs
using System;
public class main{
    public static void Main()
    {
        int n=int.Parse(Console.ReadLine());
        int temp,temp1,rem,rem1,sum=0,rev=0,r;
        temp=n;
        while(n>0)
        {
            rem=n%10;
            sum=sum+rem;
            n=n/10;
        }
        temp1=sum;
        while(sum>0)
        {
            rem1=sum%10;
            rev=rev*10+rem1;
            sum=sum/10;
        }
        r=temp1*rev;
        if(temp==r)
        {
            Console.WriteLine("Magic number");
        }
        else
        {
            Console.WriteLine("Not magic");
        }
    }
}

File: magic number.js
let n=1728,rem,rem1,sum=0,rev=0;
let temp=n;
while(n>0)
{
    rem=n%10;
    sum=(sum)+rem;
    n=Math.floor(n/10);
}
let temp1=sum;
while(sum>0)
{
    rem1=sum%10;
    rev=rev*10+rem1;
    sum=Math.floor(sum/10);
}
let r= temp1*rev;
if(temp==r)
{
    console.log("magic number");
}
else
{
    console.log("not magic number");
}
